File: apps/api/pages/api/health.js
// ============================================================
// GET /api/health
//
// Health check simples. Verifica se a API esta de pe e se
// consegue conectar ao banco de dados.
// ============================================================

import { Pool } from "pg";

let pool;
function getPool() {
  if (!pool) {
    pool = new Pool({ connectionString: process.env.DATABASE_URL });
  }
  return pool;
}

export default async function handler(req, res) {
  try {
    const client = await getPool().connect();
    try {
      await client.query("SELECT 1");
      res.status(200).json({ status: "ok", database: "ok" });
    } finally {
      client.release();
    }
  } catch (err) {
    res.status(500).json({ status: "ok", database: "erro", detalhe: err.message });
  }
}

File: apps/api/routers/health.py
# ============================================================
# GET /api/health
#
# Health check simples. Verifica se a API esta de pe e se
# consegue conectar ao banco de dados.
# ============================================================

from fastapi import APIRouter, Depends

from database import obter_conexao_dependency
from models import HealthResponse

router = APIRouter()


@router.get("/health", response_model=HealthResponse)
def health(conexao=Depends(obter_conexao_dependency)):
    try:
        with conexao.cursor() as cursor:
            cursor.execute("SELECT 1")
        return HealthResponse(status="ok", database="ok")
    except Exception as erro:
        return HealthResponse(status="ok", database="erro", detalhe=str(erro))
